diff --git a/examples/oauth.ts b/examples/jarm.ts
index dde3bbc..ef8729a 100644
--- a/examples/oauth.ts
+++ b/examples/jarm.ts
@@ -16,6 +16,8 @@ let redirect_uri!: string
 
 let config = await client.discovery(server, clientId, clientSecret)
 
+client.useJwtResponseMode(config)
+
 let code_challenge_method = 'S256'
 /**
  * The following (code_verifier and potentially state) MUST be generated for
