diff --git a/examples/oauth.ts b/examples/par.ts
index d87c63e..58bb51d 100644
--- a/examples/oauth.ts
+++ b/examples/par.ts
@@ -51,7 +51,7 @@ let state!: string
     parameters.state = state
   }
 
-  let redirectTo = client.buildAuthorizationUrl(config, parameters)
+  let redirectTo = await client.buildAuthorizationUrlWithPAR(config, parameters)
 
   console.log('redirecting to', redirectTo.href)
   // now redirect the user to redirectTo.href
